ldt/params/AquaCrop/define_AC_compartments.F90

```diff
@@ -0,0 +1,212 @@
+    enddo
+
+    !- Get max rooting depth
+    !- Finds crop inventory
+    call ESMF_ConfigFindLabel(LDT_config,"AquaCrop crop library directory:",rc=rc)
+    call ESMF_ConfigGetAttribute(LDT_Config, dir,rc=rc)
+    call LDT_verify(rc,"AquaCrop crop library directory: not defined")
+
+    cropinv_file = trim(dir)//"AC_Crop.Inventory"
+    !- Open file:
+    ftn = LDT_getNextUnitNumber()
+    open(ftn, file=cropinv_file, status='old', form='formatted',&
+            iostat=ios1)
+    call LDT_verify(ios1,"AquaCrop crop inventory does not exist")
+    !- Read crop inventory file:
+    read(ftn,fmt=*) num_types
+    read(ftn,fmt=*) header1
+    allocate(rootingd(num_types))
+    do i = 1, num_types
+        read(ftn,fmt=*) k, read_cropname, read_fullname 
+        ! Open corresponding crop file
+        crop_file = trim(dir)//"AC_Crop.Files/"//trim(read_cropname)//".CRO"
+        ftn1 = LDT_getNextUnitNumber()
+        open(ftn1, file=crop_file, status='old', form='formatted',&
+                iostat=ios1)
+        call LDT_verify(ios1,"AquaCrop crop file does not exist")
+        do j=1,37
+            read(ftn1,fmt=*)
+        enddo
+        read(ftn1,fmt=*) rootingd(i)
+        ! Check validity
+        if((rootingd(i).le.0).or.(rootingd(i).gt.3))then
+          write(LDT_logunit,*) "[WARN] AquaCrop rooting depth not in valid range"
+          write(LDT_logunit,*) "Program stopping ..."
+        endif
+        call LDT_releaseUnitNumber(ftn1)
+    end do    
+    call LDT_releaseUnitNumber(ftn)
+
+    ! calculate the compartment sizes for each crop type
+    ! Max 12 compartments and by default 10 cm each
+    allocate(csizes(num_types,12))
+
+
+    do ct = 1, num_types
+        ! Adjust number of compartments base don the soil profile
+        ! depth
+        ! (based on DetermineNrandThicknessCompartments from global.f90)
+        ! Does not use rootingd
+        TotalDepthL = 0.
+        do i = 1, AquaCrop_struc(n)%nlayers
+            TotalDepthL = TotalDepthL + AquaCrop_struc(n)%lthickness(i)
+        end do
+        TotDepthC = 0.
+        nrcomp = 0
+        loop: do
+            DeltaZ = (TotalDepthL - TotDepthC)
+            nrcomp = nrcomp + 1
+            if (DeltaZ > 0.10) then
+                csizes(ct,nrcomp) = 0.10
+            else
+                csizes(ct,nrcomp) = DeltaZ
+            end if
+            TotDepthC = TotDepthC + csizes(ct,nrcomp)
+            if ((nrcomp == 12) &
+                .or. (abs(TotDepthC - TotalDepthL) < 0.0001)) exit loop
+            end do loop
+
+        ! AquaCrop procedure to adjust the size of the compartments 
+        ! (based on AdjustSizeCompartments in global.f90)
+        ! Requires rootingd
+
+        ! Calculate new depth of compartments
+        TotDepthC = 0.
+        do i = 1, nrcomp
+            TotDepthC = TotDepthC + csizes(ct,i)
+        enddo
+
+        if (nrcomp < 12) then
+            loop_bis: do
```

Comment:
is this type of looping done in LIS, or would we need a "while" loop instead?

Reply:
It would be easier if we could just keep the copy-pasting of the AC source code here. In addition, I've found a few examples where these kinds of do loops are used (+ it is used many times in the AC src, which is now included in LIS).